bin/hbase-daemon.sh

```diff
@@ -376,6 +376,19 @@ case $startStop in
     wait_until_done $!
   ;;
 
+(status)
+    if [ -f $HBASE_PID ]; then
+      pid=`cat $HBASE_PID`
+      if kill -0 "$pid" > /dev/null 2>&1; then
```

Comment:
Today I learned about this `kill -0` trick, nice. :)

Reply:
IIRC, kill -0 also work for a tid, and also if the process is already dead and another process reuses the pid, this may incorrectly report the state.
IIRC for master and regionserver, we will add a `-Dproc_master` or `-Dproc_regionserver` in the start command, we could first check if the process is what we expected?

Reply:
We can leave it for now, and improve later.